src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs

```diff
@@ -27,167 +20,176 @@ public sealed class StatelessFuncAnalyzer : DiagnosticAnalyzer {
 		private static void RegisterAnalysis( CompilationStartAnalysisContext context ) {
 
 			Compilation compilation = context.Compilation;
-			ISymbol statelessFuncAttr = compilation.GetTypeByMetadataName( "D2L.CodeStyle.Annotations.Contract.StatelessFuncAttribute" );
+			ISymbol? statelessFuncAttr = compilation.GetTypeByMetadataName( "D2L.CodeStyle.Annotations.Contract.StatelessFuncAttribute" );
 
 			if( statelessFuncAttr == null ) {
 				return;
 			}
 
 			ImmutableHashSet<ISymbol> statelessFuncs = GetStatelessFuncTypes( compilation );
 
-			context.RegisterSyntaxNodeAction(
+			context.RegisterOperationAction(
 				ctx => {
 					AnalyzeArgument(
 						ctx,
+						(IArgumentOperation)ctx.Operation,
 						statelessFuncAttr,
 						statelessFuncs
 					);
 				},
-				SyntaxKind.Argument
+				OperationKind.Argument
 			);
 		}
 
 		private static void AnalyzeArgument(
-			SyntaxNodeAnalysisContext context,
+			OperationAnalysisContext context,
+			IArgumentOperation argumentOperation,
 			ISymbol statelessFuncAttribute,
 			ImmutableHashSet<ISymbol> statelessFuncs
 		) {
-			ArgumentSyntax syntax = context.Node as ArgumentSyntax;
-
-			SemanticModel model = context.SemanticModel;
-
-			IParameterSymbol param = syntax.DetermineParameter(
-				model,
-				allowParams: false,
-				context.CancellationToken
-			);
-
-			if( param == null ) {
-				return;
+			if( argumentOperation.ArgumentKind == ArgumentKind.DefaultValue ) {
+				return;
 			}
 
-			ImmutableArray<AttributeData> paramAttributes = param.GetAttributes();
-			if( !paramAttributes.Any( a => a.AttributeClass.Equals( statelessFuncAttribute, SymbolEqualityComparer.Default ) ) ) {
-				return;
+			IParameterSymbol? param = argumentOperation.Parameter;
+			if( param == null ) {
+				return;
 			}
 
-			ExpressionSyntax argument = syntax.Expression;
-
+			if( !IsParameterMarkedStateless( statelessFuncAttribute, param ) ) {
+				return;
+			}
+
+			IOperation operationToInspect = argumentOperation.Value;
+			switch( operationToInspect ) {
+				/**
+				 * Any statement defining an anonymous function:
+				 *   () => 0
+				 *   x => 0
+				 *   delegate () { return 0; }
+				 * We grab the target to get the actual function out instead of
+				 * the act of defining it.
+				 */

```

Comment:
nit: we use `//`: https://search.d2l.dev/search?project=Brightspace&full=%22%2F*%22&defs=&refs=&path=%21Specs+%22D2L.CodeStyle.Analyzers%22&hist=&type=&xrd=&nn=1&si=path&si=path

All the extra `*`s you're gonna save bytes anyway :)

Reply:
Lol I see, this file was already doing it. I don't like the inconsistency in this repo (and the rest of our code) but approved regardless.

Reply:
Went with what was (mostly) in the file, (which was admittedly me in the first place). Might follow-up.

👍